src/utils.ts

```diff
@@ -347,10 +347,17 @@ export function maybeCallback<T>(
     return promise;
   }
 
-  promise.then(
-    result => callback(undefined, result),
-    error => callback(error)
-  );
+  promise
+    .then(
+      result => callback(undefined, result),
+      error => callback(error)
+    )
+    .catch(() => {
```

Comment:
The typical result here would be to actually re-throw the exception outside of the current Promise chain, just like a callback would, e.g. inside a `process.nextTick(() => { throw exception; })`